addons/mail/static/tests/web/chatter_tests.js

```diff
@@ -770,3 +770,42 @@ QUnit.test("upload attachment on draft record", async (assert) => {
     await afterNextRender(() => dropFiles($(".o-mail-Dropzone")[0], [file]));
     await waitUntil("button[aria-label='Attach files']:contains(1)");
 });
+
+QUnit.test("composer should rerender when switching thread", async (assert) => {
+    const pyEnv = await startServer();
+    const [partnerId_1, partnerId_2] = pyEnv["res.partner"].create([
+        {
+            description: "test",
+            display_name: "Partner 1",
+        },
+        { display_name: "Partner 2" },
+    ]);
+    const views = {
+        "res.partner,false,form": `
+            <form string="Partners">
+                <sheet>
+                    <field name="name"/>
+                    <field name="description"/>
+                </sheet>
+                <div class="oe_chatter">
+                    <field name="message_ids"/>
+                </div>
+            </form>`,
+    };
+    patchUiSize({ size: SIZES.LG });
+    const { openView } = await start({ serverData: { views } });
+    await openView(
+        {
+            res_model: "res.partner",
+            res_id: partnerId_1,
+            views: [[false, "form"]],
+        },
+        { resIds: [partnerId_1, partnerId_2] }
+    );
+
+    await click("button:contains(Log note)");
+    await click(".o_pager_next");
+    assert.containsOnce($, ".o-mail-Composer-input");
```

Comment:
This assertion looks useless.

If somehow the test fails because we change the open state of composer in chatter, then it's not a problem: we just adapt test to make the click.
The assertion gives the wrong impression that it is very important for the spec, when it isn't. Only the assertion on mention after using pager is necessary